core-rust/state-manager/src/state_manager.rs

```diff
@@ -562,7 +561,7 @@ where
 
             let validate_result = self
                 .user_transaction_validator
-                .validate_and_create_executable(&parsed);
+                .validate_and_create_executable(&parsed, proposed_payload.len());
```

Comment:
I'm very worried that it's not clear what this proposed payload length is and that we might end up being inconsistent in whether we use the notarized payload length or the ledger payload length(!!).

To mitigate this, could we perhaps have `validate_and_create_executable` only take some struct which is created by `UserTransactionValidator and ensures we use the same payload length for both?

Which should probably be the notarized payload length.

However, decoding the LedgerPayload and then encoding the Notarized Payload just to subtract a constant from the length feels a little silly - so maybe we can try to be a little more clever and calculate that constant, and have some tests which verify that constant is accurate?

Reply:
Yup, totally agree here. As mentioned earlier though I think this will require a bit of a refactor and would prefer to have this PR focused on just integration. Will work on the refactor right after this though.